app/src/main/java/org/mozilla/fenix/home/HomeScreenViewModel.kt

```diff
@@ -6,8 +6,10 @@ package org.mozilla.fenix.home
 
 import android.os.Parcelable
 import androidx.lifecycle.ViewModel
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
 
 class HomeScreenViewModel : ViewModel() {
     var layoutManagerState: Parcelable? = null
     var motionLayoutProgress = 0F
+    var mode: BrowsingMode? = null
```

Comment:
Adding the last mode to the view model is to fix #5131

Reply:
If we're putting mode in the view model we should probably switch to using LiveData and put all our mode-related code in the model. [I've been planning to do this](https://github.com/mozilla-mobile/fenix/pull/4603#discussion_r312206027) but haven't gotten around to it yet.

Reply:
We eventually want to convert the entire Home Screen to the new arch as well but I just wanted to get a quick fix in for these issues https://github.com/mozilla-mobile/fenix/issues/4597